{
  "tests": [
    {
      "name": "Test advanceddesign - brownian inheritance",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestAdvancedDesign::test_brownian_inheritance",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test advanceddesign - collide doesnt call ttc",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestAdvancedDesign::test_collide_doesnt_call_ttc",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test advanceddesign - container doesnt hide vars",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestAdvancedDesign::test_container_doesnt_hide_vars",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test advanceddesign - container inheritance",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestAdvancedDesign::test_container_inheritance",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test advanceddesign - container inherits ttc",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestAdvancedDesign::test_container_inherits_ttc",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test advanceddesign - hidden variables",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestAdvancedDesign::test_hidden_variables",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - custom pos type",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_custom_pos_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - custom vel type",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_custom_vel_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - default pos type",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_default_pos_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - default vel type",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_default_vel_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - hidden vars",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_hidden_vars",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - pos list or array check",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_pos_list_or_array_check",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - pos too long check",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_pos_too_long_check",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - pos too short check",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_pos_too_short_check",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - set vel list or array check",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_set_vel_list_or_array_check",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - set vel too long",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_set_vel_too_long",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - set vel too short",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_set_vel_too_short",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - set vel type",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_set_vel_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - vel list or array check",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_vel_list_or_array_check",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - vel too long check",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_vel_too_long_check",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test ballinternals - vel too short check",
      "setup": "",
      "run": "pytest -vv tests/test_fine_details.py::TestBallInternals::test_vel_too_short_check",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test norelativeimports - balls module",
      "setup": "",
      "run": "pytest -vv tests/test_important_details.py::TestNoRelativeImports::test_balls_module",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test norelativeimports - physics module",
      "setup": "",
      "run": "pytest -vv tests/test_important_details.py::TestNoRelativeImports::test_physics_module",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test norelativeimports - simulations module",
      "setup": "",
      "run": "pytest -vv tests/test_important_details.py::TestNoRelativeImports::test_simulations_module",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - balls correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_balls_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - balls exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_balls_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - construction",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_construction",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - container correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_container_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - container exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_container_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - doesnt crash",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_doesnt_crash",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - multiballsim args",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_multiballsim_args",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - multiballsim exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_multiballsim_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - next collision exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_next_collision_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - run called",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_run_called",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task10 - setup figure exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask10::test_setup_figure_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task11 - creating hist",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask11::test_creating_hist",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task11 - doesnt crash",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask11::test_doesnt_crash",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task11 - output",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask11::test_output",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task11 - running simulation",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask11::test_running_simulation",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - doesnt crash",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_doesnt_crash",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - ke correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_ke_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - ke exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_ke_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - momentum correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_momentum_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - momentum exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_momentum_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - output",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_output",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - pressure correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_pressure_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - pressure exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_pressure_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - time correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_time_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task12 - time exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask12::test_time_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task13 - doesnt crash",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask13::test_doesnt_crash",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task13 - output",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask13::test_output",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task13 - t equipartition correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask13::test_t_equipartition_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task13 - t equipartition exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask13::test_t_equipartition_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task14 - doesnt crash",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask14::test_doesnt_crash",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task14 - output",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask14::test_output",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task14 - t ideal correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask14::test_t_ideal_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task14 - t ideal exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask14::test_t_ideal_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task15 - doesnt crash",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask15::test_doesnt_crash",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task15 - maxwell args",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask15::test_maxwell_args",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task15 - maxwell correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask15::test_maxwell_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task15 - maxwell exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask15::test_maxwell_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task15 - output",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask15::test_output",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task15 - speeds correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask15::test_speeds_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task15 - speeds exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask15::test_speeds_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task16 - curve fit called",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask16::test_curve_fit_called",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task16 - doesnt crash",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask16::test_doesnt_crash",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task16 - multiple sims created",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask16::test_multiple_sims_created",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task16 - output",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask16::test_output",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task17 - bbpositions exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask17::test_bbpositions_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task17 - browniansimulation created",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask17::test_browniansimulation_created",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task17 - browniansimulation exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask17::test_browniansimulation_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task17 - defaults init args",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask17::test_defaults_init_args",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task17 - doesnt crash",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask17::test_doesnt_crash",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task17 - initialisation args",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask17::test_initialisation_args",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - ball class exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_ball_class_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - construction",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_construction",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - init args",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_init_args",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - mass correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_mass_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - mass method exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_mass_method_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - mass type",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_mass_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - module exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_module_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - move correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_move_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - move method exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_move_method_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - pos correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_pos_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - pos method exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_pos_method_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - pos returns array",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_pos_returns_array",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - radius correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_radius_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - radius method exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_radius_method_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - radius type",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_radius_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - set vel correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_set_vel_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - set vel method exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_set_vel_method_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - set vel sets array",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_set_vel_sets_array",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - vel correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_vel_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - vel method exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_vel_method_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task2 - vel returns array",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask2::test_vel_returns_array",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task3 - move single patch",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask3::test_move_single_patch",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task3 - patch correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask3::test_patch_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task3 - patch exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask3::test_patch_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task3 - patch moves",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask3::test_patch_moves",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task3 - patch type",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask3::test_patch_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task3 - single patch",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask3::test_single_patch",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task4 - ttc correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask4::test_ttc_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task4 - ttc exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask4::test_ttc_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task4 - ttc going away",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask4::test_ttc_going_away",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task4 - ttc parallel",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask4::test_ttc_parallel",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task4 - ttc return type",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask4::test_ttc_return_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task5 - collide correct 1D",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask5::test_collide_correct_1D",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task5 - collide correct 2D",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask5::test_collide_correct_2D",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task5 - collide exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask5::test_collide_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - collide correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_collide_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - collide exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_collide_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - container args",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_container_args",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - container construction",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_container_construction",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - container exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_container_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - default args",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_default_args",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - dp tot correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_dp_tot_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - dp tot exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_dp_tot_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - sa correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_sa_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - sa exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_sa_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - ttc correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_ttc_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - ttc exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_ttc_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - ttc return type",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_ttc_return_type",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - volume correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_volume_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task6 - volume exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask6::test_volume_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task7 - next collision exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask7::test_next_collision_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task7 - next collision not implemented",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask7::test_next_collision_not_implemented",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task7 - run exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask7::test_run_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task7 - run signature",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask7::test_run_signature",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task7 - setup figure exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask7::test_setup_figure_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task7 - setup figure not implemented",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask7::test_setup_figure_not_implemented",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task7 - simulation class exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask7::test_simulation_class_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task7 - simulation not inherited",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask7::test_simulation_not_inherited",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task7 - simulations exits",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask7::test_simulations_exits",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - ball correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_ball_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - ball exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_ball_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - container correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_container_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - container exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_container_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - initialisation",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_initialisation",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - initialisation args",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_initialisation_args",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - next collision exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_next_collision_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - next collision functionality",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_next_collision_functionality",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - setup figure exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_setup_figure_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task8 - singleballsimulation exists",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask8::test_singleballsimulation_exists",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task9 - doesnt crash",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask9::test_doesnt_crash",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task9 - run called",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask9::test_run_called",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task9 - run correct",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask9::test_run_correct",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    },
    {
      "name": "Test task9 - singleballsim created",
      "setup": "",
      "run": "pytest -vv tests/test_tasks.py::TestTask9::test_singleballsim_created",
      "input": "",
      "output": "",
      "comparison": "included",
      "timeout": 300,
      "points": 1
    }
  ]
}
